libcrux-psq/src/psk_registration.rs

```diff
@@ -147,6 +147,11 @@ impl Initiator {
 
         Ok(RegisteredPsk { psk, psk_handle })
     }
+
+    /// Derives a PSK from the initiator's PQ-PrePSK, if initialized.
+    pub fn derive_unregistered_psk(&self) -> Result<Psk, Error> {
+        derive_psk(&self.k_pq)
```

Comment:
This way we have to compute the PSK at least twice.
How about we add the `psk` to the state and derive it in `send_initial_message` right away? Then this becomes a getter (e.g. `unregistered_psk`), and we can just return it in `complete_handshake`?

Also, it would be great to add it to a test to make sure it works and that the derived PSK is the same here and when completing the handshake.